### PR TITLE
add DNS servers when either IPv4 or IPv6 is configured and not using …

### PR DESCRIPTION
**Pull request by benjamreis**
…DHCP

Signed-off-by: BenjiReis <benjamin.reis@vates.fr>

This fix is taken from another PR: https://github.com/xapi-project/xcp-networkd/pull/156.

### CODE REVIEW

**Comment by benjamreis** [comment]
IDK what caused the CI to failed. Build works fine on my side.

**Comment by psafont** [comment]
CI failes becuase the script used for testing has changed how the container name is generated, and has generated a name that doesn't exist in dockerhub. I'll look into this.

**Comment by benjamreis** [comment]
Thx!

File: networkd/network_server.ml [hunk 1]
```diff
@@ -666,16 +666,16 @@ module Interface = struct
                 ) ) ->
                 update_config name c ;
                 exec (fun () ->
-                    (* We only apply the DNS settings when in static IPv4 mode
-                       to avoid conflicts with DHCP mode. * The `dns` field
+                    (* We only apply the DNS settings when not in a DHCP mode
```
Comment (lindig) [inline networkd/network_server.ml:669]: Please fix the `*` while we are here

**Review by lindig — APPROVED** [review]
Looks good to me.

**Review by robhoes — APPROVED** [review]
This looks good to me.

**Comment by robhoes** [comment]
@psafont, do we still need a CI fix?

**Comment by psafont** [comment]
It should be fixed now, but we can't trigger rebuilds. The branch builds and passes the test on my machine for what is worth.

To retrigger a travis run I think a `git amend --no-edit && git push -f` would work.

**Comment by benjamreis** [comment]
Will do!

**Comment by benjamreis** [comment]
Success! \o/

**Pull request merged**